<launch>
  <!-- Launch Parameters -->
  <arg name="cameraName" value="camera" />
  <arg name="recordData" value="false" />
  <arg name="drawMarkers" value="true" />
  <arg name="showViz" value="true" />
  
  <!-- Launch camera to image frame tf publisher -->
    <node pkg="tf" type="static_transform_publisher" name="static_tf_pub" args="0.04267084279124 0.01272343768501 -0.051384767497101 -0.658349912614842 0.562615437282137 -0.328621348695924 0.376609822944149 camera image 100" />
  
  <!-- Launch motion capture listener -->
  <include file="$(find mocap_optitrack)/launch/mocap.launch" />
  
  <!-- Launch camera driver -->
  <include file="$(find ueye_cam)/launch/rgb8.launch" />
  
  <!-- Launch tag tracker -->
  <node name="aruco" pkg="aruco_ros" type="blob_node" >
    <param name="camera" value="$(arg cameraName)" />
    <param name="markerSize" value="0.2032" />
    <param name="drawMarkers" value="$(arg drawMarkers)" />
  </node>
  
  <!-- Image frame velocity filter node -->
  <node pkg="turtlebot_ekf" type="state_filter.py" name="state_filter" ns="image" output="screen"/>
  
  <!-- Move turtlebot along a line -->
    <node pkg="switch_vis_exp" type="turtlebot_control.py" name="turtlebot_control" ns="ugv0" output="screen" />
  
  <!-- Launch world to image pose publisher -->
  <node pkg="robot_pose_publisher" type="robot_pose_publisher" name="robot_pose_publisher" output="screen" >
      <param name="map_frame" value="world" />
      <param name="base_frame" value="image" />
      <param name="publish_frequency" value="360.0" />
      <param name="is_stamped" value="true" />
      <remap from="robot_pose" to="image/pose" />
  </node>
  
  <!-- Launch Vizualization -->
  <node if="$(arg showViz)" name="rviz" pkg="rviz" type="rviz" />
  
  <!-- Record data -->
  <node if="$(arg recordData)" name="record" pkg="rosbag" type="record" args="-o /home/ncr/experiment_data/exp_aug_switch.bag /output /debug /filter /markerImage /$(arg cameraName)/camera_info /markerCenters /markers /odom" />
  
</launch>
